File: app/src/main/java/com/druger/firstandroidgame/GameObject.java
package com.druger.firstandroidgame;

import android.graphics.Rect;

/**
 * Created by druger on 04.08.2015.
 */
public abstract class GameObject {
    protected int x;
    protected int y;
    protected int dx;
    protected int dy;
    protected int width;
    protected int height;

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    public int getWidth() {
        return width;
    }

    public int getHeight() {
        return height;
    }

    public Rect getRectangle(){
        return new Rect(x, y, x+width, y+height);
    }
}

File: app/src/main/java/com/druger/firstandroidgame/objects/GameObject.kt
package com.druger.firstandroidgame.objects;

import android.graphics.Rect;

/**
 * Created by druger on 04.08.2015.
 */
abstract class GameObject {
    protected open var x: Int = 0
    protected open var y: Int = 0
    protected open var dx: Int = 0
    protected open var dy: Int = 0
    protected open var width: Int = 0
    protected open var height: Int = 0

    fun getRectangle(): Rect = Rect(x, y, x + width, y + height)
}
